### DISCUSSION TITLE
How to set custom label for connectors

### DISCUSSION THREAD

**Discussion by AntoineDuComptoirDesPharmacies** [algolia/instantsearch#5502] [Q&A]
Hi everyone,

In react-instantsearch-dom, we were able to set custom label for a given connector and get back this label in the `currentRefinements`.
See : https://github.com/algolia/instantsearch/blob/master/packages/react-instantsearch-core/src/connectors/connectToggleRefinement.js#L176

Is there a way to accomplish the same in react-instantsearch-hook ?
We notice `transformItems` method on `currentRefinements` but its too late as if multiple checkbox are applied on the same attribute, we can not distinguish them.

Yours faithfully,
LCDP

**Comment by Haroenv** [comment]
React InstantSearch Hooks/InstantSearch.js indeed doesn't have this functionality, as technically every filter can only be once in the UI. in theory you could keep a separate map of which filter matches which label that you can use in CurrentRefinements. Is that something that could work for your use case?

**Reply by AntoineDuComptoirDesPharmacies** [reply]
Yes we gonna do something like this in `CurrentRefinements` and it will work until we have multiple input for same attribute. 🤞 

```
[...]
const translations = [
  ['owner.nickname', 'Vendeur'],
  ['product.laboratory.name', 'Laboratoire'],
  ['product.types', 'Type'],
  [...]
];

const computeRefinementItemLabel = (item, refinement) => {
  const prefix = item.label;
  const value = refinement.label;
  return translate(`${prefix}: ${value}`);
};
[...]
```
